MimeKit/Cryptography/BouncyCastleCertificateExtensions.cs

```diff
@@ -195,7 +195,7 @@ internal static string AsHex (this byte[] blob)
 			var hex = new ValueStringBuilder (blob.Length * 2);
 
 			for (int i = 0; i < blob.Length; i++)
-				hex.Append (blob[i].ToString ("x2"));
+				hex.AppendInvariant (blob[i], "x2");
```

Comment:
Note, we also fixed the format overload in ValueStringBuilder. This uses ISpanFormattable on .NET6.0+